From e156654d4eb2fdeb524e6b40838767a5dc918966 Mon Sep 17 00:00:00 2001
From: David Vrabel <david.vrabel@citrix.com>
Date: Thu, 18 Jun 2015 14:54:25 +0200
Subject: [PATCH] evtchn: remove the locking when unmasking an event channel

The event channel lock is no longer required to check if the port is
valid.

Signed-off-by: David Vrabel <david.vrabel@citrix.com>
diff --git a/xen/common/event_channel.c b/xen/common/event_channel.c
index ed532e9..d4e86e8 100644
--- a/xen/common/event_channel.c
+++ b/xen/common/event_channel.c
@@ -934,8 +934,6 @@ int evtchn_unmask(unsigned int port)
     struct domain *d = current->domain;
     struct evtchn *evtchn;
 
-    ASSERT(spin_is_locked(&d->event_lock));
-
     if ( unlikely(!port_is_valid(d, port)) )
         return -EINVAL;
 
@@ -1102,9 +1100,7 @@ long do_event_channel_op(int cmd, XEN_GUEST_HANDLE_PARAM(void) arg)
         struct evtchn_unmask unmask;
         if ( copy_from_guest(&unmask, arg, 1) != 0 )
             return -EFAULT;
-        spin_lock(&current->domain->event_lock);
         rc = evtchn_unmask(unmask.port);
-        spin_unlock(&current->domain->event_lock);
         break;
     }
 
